packages/journey-manager/src/index.ts

```diff
@@ -0,0 +1,166 @@
+import { type Config, create } from "@nlxai/voice-compass";
+import { find, decode, type Query, type EncodedQuery } from "./queries";
+
+/**
+ * Step ID
+ */
+export type StepId = string;
+
+/**
+ * URL match condition
+ */
+export interface UrlCondition {
+  /**
+   * Condition operator
+   */
+  operator: "eq" | "neq" | "suffix" | "prefix" | "contains" | "not_contains";
+  /**
+   * Condition value
+   */
+  value: string;
+  // TODO: add regex match condition
+}
+
+/**
+ * A single trigger
+ */
+export interface Trigger {
+  /**
+   * Event
+   */
+  event: "pageLoad" | "click";
+  /**
+   * A query identifying the element
+   */
+  query?: EncodedQuery;
+  /**
+   * URL condition
+   */
+  urlCondition?: UrlCondition;
+}
```

Comment:
Probably nicer as a discriminated union as "pageLoad" doesn't have a query and "click" always has a query:

```suggestion
export type Trigger = {
  /**
   * Event
   */
  event: "pageLoad";
  /**
   * URL condition
   */
  urlCondition?: UrlCondition;
} | {
/**
   * Event
   */
  event: "click";
  /**
   * A query identifying the element
   */
  query: EncodedQuery;
  /**
   * URL condition
   */
  urlCondition?: UrlCondition;
}
```